### PR TITLE
docs(react-datepicker-compat, public-docsite-v9): Adding DatePicker to the public docs and cleaning current docs

### PR DESCRIPTION
**Pull request by sopranopillow**
This PR does the following:
* Add DatePicker to the public docsite as a Preview Component.
* Update docs to remove Calendar since Calendar will not be exposed in our API.
* Cleanup current docs to make them simpler to follow and be more concise.

## Related Issue(s)

- Related #26244

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/77bb9ec1eeedef91315dee3750bd06b15168c693'>77bb9ec1eeedef91315dee3750bd06b15168c693</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv8 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv8"/>

**Comment by fabricteam** [comment]
## 🕵 FluentUI-v0 No visual regressions between this PR and main
<div id="vrtCommentFluentUI-v0"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluentuiv9"/>

File: packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerDisabled.stories.tsx [hunk 1]
```diff
@@ -1,36 +1,23 @@
...
-    </div>
+    <DatePicker
+      disabled
+      label="Disabled (with label)"
+      placeholder="Select a date..."
+      aria-label="Select a date"
+      className={styles.control}
+    />
   );
 };
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerDisabled.stories.tsx:23]:
```suggestion
};

Disabled.parameters = {
  docs: {
    description: {
      story:
        'DatePicker can be disabled to restrict user interaction with it.'
    },
  },
};
```

File: packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerTextInput.stories.tsx [hunk 2]
```diff
@@ -17,33 +16,20 @@ const useStyles = makeStyles({
...
     </div>
   );
 };
+
+TextInput.parameters = {
+  docs: {
+    description: {
+      story:
+        'A DatePicker supports user input. Clicking the input field will open the DatePicker, and clicking the' +
+        'field again will dismiss the DatePicker and allow text input. When using keyboard navigation (tabbing into' +
```
Comment (khmakoto) [inline packages/react-components/react-datepicker-compat/stories/DatePicker/DatePickerTextInput.stories.tsx:31]:
```suggestion
        ' field again will dismiss the DatePicker and allow text input. When using keyboard navigation (tabbing into' +
```

**Review by khmakoto — APPROVED** [review]

**Review by Hotell — APPROVED** [review]

File: apps/public-docsite-v9/.storybook/main.js [hunk 3]
```diff
@@ -21,6 +21,7 @@ module.exports = /** @type {Omit<import('../../../.storybook/main'), 'typescript
     ...getPackageStoriesGlob({ packageName: '@fluentui/react-components', callerPath: __dirname }),
```
Comment (fabricteam) [inline apps/public-docsite-v9/.storybook/main.js:21]: ## [🕵 fluentuiv8 Open the Visual Regressions report to inspect the 2 screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-fluent/index.html?organization=uifabric&project=fabricpublic&prId=27289&commitId=62056ba0b78fa7d53e3823ac589b3ab040ce1b0e&env=prod)

✅ There was 0 screenshots added, 2 screenshots removed, 1045 screenshots unchanged, 0 screenshots with different dimensions and 0 screenshots with visible difference.


  <details><summary><b>unknown</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Pivot - Overflow.Tabs - RTL.Narrow - Last tab selected.chromium.png|0| Removed|
|Pivot - Overflow.Tabs - RTL.Narrow - Overflow menu.chromium.png|0| Removed|

</details>


<div id="vrtCommentfluentuiv8"/>

**Pull request merged**